custom_components/epson_workforce/api.py

```diff
@@ -1,173 +1,93 @@
-        info_div = self.soup.find("div", class_="information")
-        if not info_div:
-            return None
-
-        # Prefer a <span> inside <p.clearfix>, else any <span> within .information
-        p = info_div.find("p", class_="clearfix")
-        span = (p.find("span") if p else None) or info_div.find("span")
-        if not span:
-            return None
-
-        status = span.get_text(strip=True)
-        return self._clean_status(status)
-
-    def _get_printer_status(self) -> str:
-        """Get printer status using multiple parsing strategies."""
-        try:
-            # Try fieldset structure first: <fieldset id="PRT_STATUS">
-            # <ul>...</ul></fieldset>
-            status = self._get_fieldset_status()
-            if status:
-                return status
-
-            # Information div structure: <div class="information"><p class="clearfix">
-            # <span>Available.</span></p></div>
-            status = self._get_information_div_status()
-            if status:
-                return status
-        except Exception:
-            return "Unknown"
-        else:
-            return "Unknown"
-
-    def update(self):
-        """Fetch the HTML page."""
+    def mac_address(self) -> str | None:
+        """Returns the MAC address of the device if available."""
+        self._ensure_parsed()
+        return (self._data or {}).get("mac_address")
+
+    def update(self) -> None:
+        """
+        Fetch and parse the HTML page from the device (rebuilds parser + resets cache).
+        """
         try:
             context = ssl._create_unverified_context()
             with urllib.request.urlopen(self._resource, context=context) as response:
-                data = response.read()
-                response.close()
-
-            self.soup = BeautifulSoup(data, "html.parser")
+                data_bytes = response.read()
+            html_text = data_bytes.decode("utf-8", errors="ignore")
+            self._parser = EpsonHTMLParser(html_text, source=self._resource)
             self.available = True
-            self._extract_device_info()
+            self._data = None  # invalidate cache
         except Exception:
             self.available = False
+            self._parser = None
+            self._data = None
+
+    def get_sensor_value(self, sensor: str) -> int | str | None:
+        """Retrieves the value of a specified sensor from the parsed printer data."""
+        self._ensure_parsed()
+        data = self._data or {}
+
+        s = (sensor or "").strip().lower()
+        if s == "printer_status":
+            return data.get("printer_status") or "Unknown"
+
+        if s == "clean":
+            return data.get("maintenance_box")
+
+        label_map = {
+            "black": "BK",
+            "cyan": "C",
+            "magenta": "M",
+            "yellow": "Y",
+            "photoblack": "PB",
+            "lightcyan": "LC",
+            "lightmagenta": "LM",
```

Comment:
@alexbbt, none of the captured status pages have "LC" or "LM" levels. Are these legit?

Reply:
They totally might not be, they were just in the original library, fair to remove them until we know we need them. I bet they are on the bigger workforce printers.

This is a great cleanup!